packages/openmrs/src/Adaptor.js

```diff
@@ -319,52 +291,54 @@ export function createPatient(data, callback = false) {
  * @param {function} [callback] - Optional callback to handle the response
  * @returns {Operation}
  */
-export function getEncounter(uuid, callback = false) {
-  return state => {
-    Log.info(`Searching for encounter with uuid: ${uuid}`);
-    const { instanceUrl } = state.configuration;
-
-    const url = `${instanceUrl}/ws/rest/v1/encounter/${uuid}`;
-
-    return agent
-      .get(url)
-      .accept('json')
-      .then(response => {
-        Log.success(`Found an encounter.`);
-
-        return handleResponse(response, state, callback);
-      })
-      .catch(handleError);
+export function getEncounter(uuid, callback = s => s) {
+  return async state => {
+    const [resolvedUuid] = expandReferences(state, uuid);
+    console.log(`Fetching encounter with UUID: ${resolvedUuid}`);
+
+    const response = await request(
+      state,
+      'GET',
+      `/ws/rest/v1/encounter/${resolvedUuid}`
+    );
+
+    console.log(
+      `Successfully retrieved for encounter with UUID: ${resolvedUuid}`
+    );
+
+    return prepareNextState(state, response, callback);
   };
 }
 
 /**
  * Gets encounters matching params
  * @example
- * getEncounters({patient: "123", fromdate: "2023-05-18"})
+ * getEncounters({ patient: "123", fromdate: "2023-05-18" })
  * @function
  * @param {object} query - Object for the patient
  * @param {function} [callback] - Optional callback to handle the response
  * @returns {Operation}
  */
-export function getEncounters(query, callback = false) {
-  return state => {
-    const qs = expandReferences(query)(state);
-    const { instanceUrl } = state.configuration;
-    const url = `${instanceUrl}/ws/rest/v1/encounter`;
-
-    Log.info(`Searching for encounters: ${JSON.stringify(qs, null, 2)}`);
-
-    return agent
-      .get(url)
-      .accept('json')
-      .query(qs)
-      .then(response => {
-        Log.success(`Found an encounter.`);
-
-        return handleResponse(response, state, callback);
-      })
-      .catch(handleError);
+export function getEncounters(query, callback = s => s) {
+  return async state => {
+    const [resolvedQuery] = expandReferences(state, query);
+    console.log('Fetching encounters by query', resolvedQuery);
+
+    const response = await request(
+      state,
+      'GET',
+      `/ws/rest/v1/encounter/`,
+      {},
+      resolvedQuery
+    );
+
+    // TODO - we could also argue that this should NOT throw.
+    if (response.body.results.length == 0) {
```

Comment:
Yeah I don't think we should throw here. Zero results is an empty array, not an exception.

Also this is a change in behavior from the latest release, which we don't really want in this PR.

Reply:
Well technically that's a change in behaviour so it's scope creep :thinking: 

Looking at the code before and after though, I think you're right. We used to say `found an encounter` (which wouldn't be true if `results.length === 0`).

So yes let's say `Found ${response.body.results.length}} results` - that's an acceptable change in the spirit of this PR.